### PR TITLE
logit invlogit

### PR DESCRIPTION
**Pull request by krpaulson**

### CODE REVIEW

**Comment by krpaulson** [comment]
Should the scaling be the opposite? i.e. scale logit input and invlogit output

File: R/logit_utils.R [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#' @title Logit and inverse logit functions
+#'
+#' @description Logit and inverse logit functions, with options to scale input
+#'   to logit, scale output of inverse logit, or perform a transformation
+#'   such that the domain in non-logit space is as specified (standard
+#'   is (0, 1)).
+#'
+#' @param x \[`numeric()`\]\cr Value to take the logit or inverse logit of.
+#' @param scale \[`numeric()`\]\cr Value to scale by. Default 1.
```
Comment (chacalle) [inline R/logit_utils.R:9]: "Value to scale `x` by"

File: R/logit_utils.R [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+#'
+#' @details
+#' Standard:
+#'
+#' * logit (x) = log (x / (1-x))
+#' * inverse logit (x) = exp (x) / (1 + exp (x))
+#'
+#' Optional scaled logit with 'scale' argument:
+#'
+#' * logit (x, s) = log (s * x / (1 - s * x))
```
Comment (chacalle) [inline R/logit_utils.R:25]: I think this would clearer to use the same notation as the transformed domain section. Took me a second to realize this

```
* logit (x, s) = log (x' / (1 - x')) where x' = x * s
* inverse logit (x, s) = exp (x') / (1 + exp (x')) where x' = x * s
```

File: R/logit_utils.R [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+#'   will return NaN and result in a warning from `logit` function.
+#'
+#' @examples
+#' # Standard
+#' logit(0.1)
+#' invlogit(-2)
+#'
+#' # Scaled
+#' x <- c(10, 20, 30)
+#' logit_x <- logit(x, scale = 0.01)
```
Comment (chacalle) [inline R/logit_utils.R:43]: So this is equivalent to `logit(x, domain_upper = 100)`. When would we actually want to use the `scale` argument, for percentages? I wonder if its clearer to just have `domain_lower` and `domain_upper` arguments?

Also does there need to be a check whether someone tries to use all these arguments?

File: tests/testthat/test-logit_utils.R [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  x <- c(10, 20, 30)
+  logit_x <- logit(x, scale = 0.01)
+  x_again <- invlogit(logit_x, scale = 0.01)
+  testthat::expect_equal(x, x_again)
+
+  # domain shift
+  x <- runif(n = 100, min = 10, max = 20)
+  logit_x <- logit(x, domain_lower = 10, domain_upper = 20)
+  x_again <- invlogit(logit_x, domain_lower = 10, domain_upper = 20)
+  testthat::expect_equal(x, x_again)
```
Comment (chacalle) [inline tests/testthat/test-logit_utils.R:21]: This is a good way to check 👍

File: R/logit_utils.R [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+
+# ===================================================================
+#' @rdname logit_utils
+#' @export
+invlogit <- function(x, scale = 1, domain_lower = 0, domain_upper = 1) {
+
+  assertive::assert_is_numeric(x)
+  assertive::assert_is_numeric(scale)
+  assertthat::assert_that(domain_lower < domain_upper)
```
Comment (chacalle) [inline R/logit_utils.R:76]: Should also check that `scale`, `domain_lower`, `domain_upper` are length one numerics

```
assertthat::assert_that(domain_lower < domain_upper,
                                       assertthat::is.number(domain_lower),
                                       assertthat::is.number(domain_upper))
```

File: DESCRIPTION [hunk 6]
```diff
@@ -18,7 +18,7 @@ Encoding: UTF-8
 LazyData: true
 ByteCompile: true
 Suggests: 
-    testthat,
+    testthat (>= 2.1.0),
```
[the remark this answers is not in the document]
Comment (krpaulson) [reply DESCRIPTION:21]: No I don't, should I try getting rid of it and see what happens?

File: R/logit_utils.R (hunk 3 above)
Comment (krpaulson) [reply R/logit_utils.R:43]: This is why I was asking folks about how they use scaled logits. As far as I can tell we mostly use the domain change, but I don't mind keeping `scale` in there as an option to make these functions more flexible.

Comment (krpaulson) [reply R/logit_utils.R:43]: Also: you could technically specify both `scale` and `domain_lower` / `domain_upper`.

File: R/logit_utils.R (hunk 5 above)
Comment (krpaulson) [reply R/logit_utils.R:76]: Or not, and leave this open to the rare scenario where you may want to apply different scaling across a vector of `x`?

File: DESCRIPTION (hunk 6 above)
[the remark this answers is not in the document]
Comment (chacalle) [reply DESCRIPTION:21]: Probably, would be nice to not have added version requirements if they aren't needed

File: R/logit_utils.R (hunk 5 above)
Comment (chacalle) [reply R/logit_utils.R:76]: Makes sense, should probably check the length is the same as `x` or length 1 then

File: R/logit_utils.R (hunk 3 above)
Comment (chacalle) [reply R/logit_utils.R:43]: I guess I was just thinking that it took me a while to figure out what was going on with the two options so might be clearer to just have the one option that seems to be flexible for both use cases. 

`scale` isn't really doing much anyway where someone could do that before hand if it was really needed.
